src/test/java/uk/gov/ons/ctp/integration/envoycuc/mockclient/MockClient.java

```diff
@@ -0,0 +1,292 @@
+      }
+      final String[] keyConstituents = requestKey.split("-");
+      final String keyType = keyConstituents[keyConstituents.length - 1];
+
+      String valueToRecord = "";
+      if (keyType.equals("UPRN")) {
+        valueToRecord = f.getUprn().getValue() + "";
+      }
+      if (keyType.equals("IP")) {
+        valueToRecord = f.getIpAddress();
+      }
+      if (keyType.equals("TELNO")) {
+        valueToRecord = f.getTelNo();
+      }
+
+      final List<Integer> postingsList =
+          postedMap.containsKey(valueToRecord) ? postedMap.get(valueToRecord) : new ArrayList<>();
+
+      final Date now = new Date(System.currentTimeMillis());
+      final Date oneHourAgo = DateUtils.addHours(now, -1);
+      SimpleDateFormat dmformatter = new SimpleDateFormat("DDDHH");
+      int dayHourNow = Integer.parseInt(dmformatter.format(now));
+      int postsWithinScopeCount = 0;
+      for (int postDateHour : postingsList) {
+        if (postDateHour == dayHourNow) {
+          postsWithinScopeCount++;
+        }
+      }
+      final String recordKey = requestKey + "(" + valueToRecord + ")";
+      recordRequest(isValidRequest, recordKey, noRequestAllowed, postsWithinScopeCount);
+    }
+    return isValidRequest;
+  }
+
+  private void recordRequest(
+      final IsValidRequest isValidRequest,
+      final String recordKey,
+      final int noRequestAllowed,
+      final int postsWithinScopeCount) {
+    final LimitStatus limitStatus = new LimitStatus();
+    limitStatus.setCode(recordKey);
+    final CurrentLimit currentLimit = new CurrentLimit();
+    currentLimit.setRequestsPerUnit(noRequestAllowed);
+    currentLimit.setUnit("HOUR");
+    limitStatus.setCurrentLimit(currentLimit);
+    if (postsWithinScopeCount >= noRequestAllowed) {
+      isValidRequest.setValid(false);
+      limitStatus.setLimitRemaining(0);
+    } else {
+      limitStatus.setLimitRemaining(noRequestAllowed - postsWithinScopeCount);
+    }
+    isValidRequest.getLimitStatusList().add(limitStatus);
+  }
+
+  private void postRequest(List<String> requestKeyList, final RateLimiterClientRequest request) {
+    for (String requestKey : requestKeyList) {
+      Map<String, List<Integer>> postedMap = postingsTimeMap.get(requestKey);
+      if (postedMap == null) {
+        return;
+      }
+      final String[] keyConstituents = requestKey.split("-");
+      final String keyType = keyConstituents[keyConstituents.length - 1];
+      String postingsListKey = null;
+      if (keyType.equals("UPRN")) {
+        postingsListKey = request.getUprn().getValue() + "";
+      }
+      if (keyType.equals("IP")) {
+        postingsListKey = request.getIpAddress();
+      }
+      if (keyType.equals("TELNO")) {
+        postingsListKey = request.getTelNo();
+      }
+      postedMap.computeIfAbsent(postingsListKey, k -> new ArrayList<>());
+      final Date now = new Date(System.currentTimeMillis());
+      SimpleDateFormat dmformatter = new SimpleDateFormat("DDDHH");
```

Comment:
Consider factoring this date formatting out into a helper method since you have used it elsewhere